sklearn/cluster/bicluster.py

```diff
@@ -156,6 +169,8 @@ def _svd(self, array, n_components, n_discard):
 
         assert_all_finite(u)
         assert_all_finite(vt)
+        if u.shape[1] == 1 and vt.shape[0] == 1:
```

Comment:
> Who's the expert on these algorithms?

Well, they were implemented by @kemaleren, and it looks like @larsmans attempted some cleanup, but I've had the impression for a while that they are somewhat orphaned, given how long issues like #2484 have been around.